examples/Example14/include/Example14EMShowerModel.hh

```diff
@@ -0,0 +1,112 @@
+// *                                                                  *
+// * The  Geant4 software  is  copyright of the Copyright Holders  of *
+// * the Geant4 Collaboration.  It is provided  under  the terms  and *
+// * conditions of the Geant4 Software License,  included in the file *
+// * LICENSE and available at  http://cern.ch/geant4/license .  These *
+// * include a list of copyright holders.                             *
+// *                                                                  *
+// * Neither the authors of this software system, nor their employing *
+// * institutes,nor the agencies providing financial support for this *
+// * work  make  any representation or  warranty, express or implied, *
+// * regarding  this  software system or assume any liability for its *
+// * use.  Please see the license in the file  LICENSE  and URL above *
+// * for the full disclaimer and the limitation of liability.         *
+// *                                                                  *
+// * This  code  implementation is the result of  the  scientific and *
+// * technical work of the GEANT4 collaboration.                      *
+// * By using,  copying,  modifying or  distributing the software (or *
+// * any work based  on the software)  you  agree  to acknowledge its *
+// * use  in  resulting  scientific  publications,  and indicate your *
+// * acceptance of all terms of the Geant4 Software license.          *
+// ********************************************************************
+//
+#ifndef PAR03EMSHOWERMODEL_HH
+#define PAR03EMSHOWERMODEL_HH
+
+#include <unordered_map>
+#include "G4VFastSimulationModel.hh"
+//#include <AdePT/ArgParser.h>
+#include <CopCore/SystemOfUnits.h>
+#include "AdeptIntegration.h"
+
+#include <G4HepEmData.hh>
+#include <G4HepEmParameters.hh>
+#include <G4HepEmState.hh>
+#include <G4HepEmStateInit.hh>
+
+class Example14EMShowerMessenger;
+class G4FastSimHitMaker;
+class G4VPhysicalVolume;
+
+/**
+ * @brief Example fast simulation model for EM showers.
+ *
+ * Parametrisation of electrons, positrons, and gammas. It is triggered if those
+ * particles enter the detector so that there is sufficient length for the
+ * shower development (max depth, controlled by the UI command).
+ *
+ * Using AdePT
+ */
+
+class Example14EMShowerModel : public G4VFastSimulationModel {
+public:
+  Example14EMShowerModel(G4String, G4Region *);
+  Example14EMShowerModel(G4String);
+  ~Example14EMShowerModel();
+
+  /// There are no kinematics constraints. True is returned.
+  virtual G4bool ModelTrigger(const G4FastTrack &) final;
+  /// Model is applicable to electrons, positrons, and photons.
+  virtual G4bool IsApplicable(const G4ParticleDefinition &) final;
+
+  /// Take particle out of the full simulation (kill it at the entrance
+  /// depositing all the energy). Simulate the full shower using AdePT library.
+  virtual void DoIt(const G4FastTrack &, G4FastStep &) final;
+
+  virtual void Flush() final;
```

Comment:
Should be `override` instead of `virtual` - compilation should hard-error if the necessary interfaces aren't there!